kbs/src/plugins/resource/backend.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2023 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use serde::Deserialize;
+
+use super::local_fs;
+use super::{ResourceError as Error, ResourceResult as Result};
+
+type RepositoryInstance = Arc<dyn StorageBackend>;
+
+/// Interface of a `Repository`.
+#[async_trait::async_trait]
+pub trait StorageBackend: Send + Sync {
+    /// Read secret resource from repository.
+    async fn read_secret_resource(&self, resource_desc: ResourceDesc) -> Result<Vec<u8>>;
+
+    /// Write secret resource into repository
+    async fn write_secret_resource(&self, resource_desc: ResourceDesc, data: &[u8]) -> Result<()>;
+}
+
+#[derive(Debug, Clone)]
+pub struct ResourceDesc {
+    pub repository_name: String,
+    pub resource_type: String,
+    pub resource_tag: String,
+}
+
+impl TryFrom<&str> for ResourceDesc {
+    type Error = Error;
+
+    fn try_from(value: &str) -> Result<Self> {
+        let path_parts: Vec<&str> = value.split('/').collect();
+        let resource_description = match path_parts.len() {
+            2 => Self {
+                repository_name: "default".into(),
+                resource_type: path_parts[0].into(),
+                resource_tag: path_parts[1].into(),
+            },
+            3 => Self {
+                repository_name: path_parts[0].into(),
+                resource_type: path_parts[1].into(),
+                resource_tag: path_parts[2].into(),
+            },
+            _ => return Err(Error::ParseResourceDescription),
+        };
+
+        if path_parts[0] == "."
+            || path_parts[0] == ".."
+            || path_parts[1] == "."
+            || path_parts[1] == ".."
+        {
+            return Err(Error::MalwaredResourceDescription);
```

Comment:
I would just call this `IllegalResourceDesciprtion`. This could be the result of user error. It isn't necessarily malware.

Reply:
Sounds reasonable. Let me fix this.